src/server/routes/fileProcessing.js

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const express = require('express');
+const Reading = require('../models/Reading');
+const moment = require('moment');
+const streamBuffers = require('stream-buffers');
+const multer = require('multer');
+const streamToDB = require('../services/loadFromCsvStream');
+const authenticator = require('./authenticator');
+
+const router = express.Router();
+
+// The upload here ensures that the file is saved to server RAM rather than disk
+const upload = multer({ storage: multer.memoryStorage() });
+
+router.use(authenticator);
+router.post('/:meter_id', upload.single('csvFile'), async (req, res) => {
+	const id = parseInt(req.params.meter_id);
```

Comment:
You might want to put this line inside the try. If you get a POST to `/notANumber` instead of a number, you'll get weird errors. Also maybe use `Number()` instead of `parseInt`. It does less crazy things when you don't get a number.

Reply:
My latest commit has parseInt in try block.